docs/snakefiles/rules.rst

```diff
@@ -433,6 +433,27 @@ Apart from Python scripts, this mechanism also allows you to integrate R_ and R
 
 In the R script, an S4 object named ``snakemake`` analog to the Python case above is available and allows access to input and output files and other parameters. Here the syntax follows that of S4 classes with attributes that are R lists, e.g. we can access the first input file with ``snakemake@input[[1]]`` (note that the first file does not have index ``0`` here, because R starts counting from ``1``). Named input and output files can be accessed in the same way, by just providing the name instead of an index, e.g. ``snakemake@input[["myfile"]]``.
 
+Furthermore, one can integrate Jupyter_ Notebooks as follows:
+
+.. _Jupyter: https://jupyter.org/
+
+.. code-block:: python
+
+    rule NAME:
+        input:
+            "path/to/inputfile",
+            "path/to/other/inputfile"
+        output:
+            "path/to/outputfile",
+            "path/to/another/outputfile",
+            notebook_output = "path/to/processed_notebook.ipynb"
+        script:
```

Comment:
I would like to have a separate directive for this, called `notebook`:
```suggestion
        notebook:
```

This provides more clarity with the other formats that are allowed below.

Reply:
Is the idea for the `script` and `notebook` directives to be mutually exclusive, or would they be executed in some order?

Reply:
Mutually exclusive. Like shell and script.

Reply:
It requires some knowledge of the Snakemake parser. I can do that part if you want.

Reply:
Yes, that would be nice :)